test/extended/openstack/loadbalancers.go

```diff
@@ -0,0 +1,239 @@
+
+		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+		jig.Labels = labels
+		svc, err := jig.CreateLoadBalancerService(120*time.Second, func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: 8082, TargetPort: intstr.FromInt(8081)}}
+			svc.Spec.Selector = labels
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Checks from openshift perspective")
+		o.Expect(svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
+			"load-balancer-id annotation missing")
+		loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+		o.Expect(svc.Status.LoadBalancer.Ingress[0].IP).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+		o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeCluster),
+			"Unexpected ExternalTrafficPolicy on svc specs")
+
+		g.By("Checks from openstack perspective")
+		lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(lb.Provider).Should(o.Equal("amphora"), "Unexpected provider in the Openstack LoadBalancer")
+
+		svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+		svcPort := fmt.Sprint(svc.Spec.Ports[0].Port)
+		fip, err := getFipbyFixedIP(networkClient, lb.VipAddress)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(fip.FloatingIP).Should(o.Equal(svcIp), "Unexpected floatingIp in the Openstack LoadBalancer")
+		o.Expect(lb.Pools).Should(o.HaveLen(1), "Unexpected number of pools on Openstack LoadBalancer %q", lb.Name)
+
+		pool, err := pools.Get(loadBalancerClient, lb.Pools[0].ID).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(pool.Protocol).Should(o.Equal("UDP"), "Unexpected protocol on Openstack LoadBalancer Pool: %q", pool.Name)
+		o.Expect(pool.ProvisioningStatus).Should(o.Equal("ACTIVE"), "Unexpected Provisioning Status on Openstack Pool: %q", pool.Name)
+		//Set as OFFLINE in vexxhost despite the lb is operative
+		//o.Expect(pool.OperatingStatus).Should(o.Equal("ONLINE"), "Unexpected Operating Status on Openstack Pool: %q", pool.Name)
+
+		lbMethod, err := getClusterLoadBalancerSetting("lb-method", cloudProviderConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(pool.LBMethod).Should(o.Equal(lbMethod), "Unexpected LBMethod on Openstack Pool: %q", pool.LBMethod)
+
+		g.By("accessing the service 100 times from outside and storing the name of the pods answering")
+		time.Sleep(10 * time.Minute) // Give time to the cluster to be ready
+		results := make(map[string]int)
+		for i := 0; i < 100; i++ {
+			// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+			podName, err := getPodNameThroughUdpLb(svcIp, svcPort, "hostname")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if val, ok := results[podName]; ok {
+				results[podName] = val + 1
+			} else {
+				results[podName] = 1
+			}
```

Comment:
You shouldn't need to initialize every element, check out this: https://go.dev/play/p/9VCO-jzrDX7

Reply:
Fantastic! Thank you!